exercises/chapter11/test/Main.purs

```diff
@@ -2,10 +2,108 @@ module Test.Main where
 
 import Prelude
 
+import Test.MySolutions
+import Test.NoPeeking.Solutions  -- Note to reader: Delete this line
+
 import Effect (Effect)
-import Effect.Class.Console (log)
+import Control.Monad.Writer (runWriterT, execWriter, tell)
+import Control.Monad.Except (runExceptT)
+import Control.Monad.State (runStateT, execState)
+import Data.Either (Either(..))
+import Data.Monoid.Additive (Additive(..))
+import Data.Newtype (unwrap)
+import Data.Tuple (Tuple(..))
+import Test.Unit (TestSuite, success, suite, test)
+import Test.Unit.Assert as Assert
+import Test.Unit.Main (runTest)
 
 main :: Effect Unit
-main = do
-  log "🍝"
-  log "You should add some tests."
+main =
+  runTest do
+    test "" success
+    {-  Move this block comment starting point to enable more tests
+Note to reader: Delete this line to expand comment block -}
+    suite "Exercises Group - The State Monad" do
+      suite "testParens" do
+        test "Empty String" do
+          Assert.equal true
+            $ testParens ""
+        test "(()(())())" do
+          Assert.equal true
+            $ testParens "(()(())())"
+        test ")" do
+          Assert.equal false
+            $ testParens ")"
+        test "(()()" do
+          Assert.equal false
+            $ testParens "(()()"
+        test ")(" do
+          Assert.equal false
+            $ testParens ")("
```

Comment:
What about a convenience wrapper for these tests to avoid typos when repeating the input as the test name?
For example, allow calling like this:
```purs
testParens true "(()(())())"
testParens false "(()()"
```

Reply:
fixed